backend/src/main.rs

```diff
@@ -616,8 +629,30 @@ async fn user_connected(ws: WebSocket, games: Games, stats: Arc<Mutex<InMemorySt
                     }
                 }
                 UserMessage::Action(m) => {
+                    let user_single_game_session;
+                    if let interactive::Message::SetUserMultiGameSessionPolicy(
+                        game_state::UserMultiGameSessionPolicy::SingleSessionOnly,
+                    ) = m
+                    {
+                        user_single_game_session = true;
+                    } else {
+                        user_single_game_session = true;
+                    }
+
                     match game.game.interact(m, player_id, &logger) {
                         Ok(msgs) => {
+                            if user_single_game_session {
```

Comment:
hm, why do we need to check this on every interaction? Dropping existing users when a new one joins seems like it should be sufficient, right?

Reply:
this check is only needed when the action is SetUserMultiGameSessionPolicy with SingleSessionOnly. We have to scan the users to reduce existing users with multiple sessions to just one. The ideal place will be inside either interactive or game_state, but then we have to pass down users, which is only defined in main.rs. Please let me know how you'd like to handle. (We can also just not do anything at all to make the code look cleaner if u'd like)

Reply:
also, any reason why users is not part of the game state?

Reply:
Websocket connections aren't part of the game state because the `shengji-core` library is intended to be agnostic to how the players actually interact with the backend. The `backend` code does the appropriate mapping in both directions to make it work.

I'd prefer not to do this check here, and instead update the phrasing on the setting in the UI to say something like

'allow players to be shadowed by joining with the same name"

Reply:
check is removed as well